hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -96,6 +97,17 @@ String getSasToken() {
     return sasToken;
   }
 
+  public ListResultSchema getListResultSchema()
+      throws AzureBlobFileSystemException {
+    if (result instanceof AbfsHttpConnection) {
+      return ((AbfsHttpConnection) this.result).getListResultSchema();
+    } else {
+      throw new AbfsRestOperationException(-1, null,
```

Comment:
Do we need to throw an exception here?

By default, when this field is unset, we would just return null.
Should we follow similar behaviour?

Do we expect this exception to ever be thrown?

Reply:
Result here is the response from store backend and List calls will only be supported HttpConnection. Is more of a safe guard check, though it can never get into else case. 
If it does for any reason, we will want the workload to fail immediately highlighting the reason explicitly for easy debugging.